### PR TITLE
Make qfile also check the prune lib registry

### PR DESCRIPTION
**Pull request by sbesl**
See bug #677982.

After portage unmerges a package, it may still keep library files that are used by other packages. These files however are not recorded in any CONTENTS files. Portage instead records such files in the prune lib registry. [1]
This pull request adds functionality to make qfile also check the prune lib registry when searching for owners of a file. This further completes the information presented to the user about which files are portage-managed.

[1] Actually, portage not only records otherwise unmanaged files in the plib registry, but also library files that got re-added to CONTENTS files after a version upgrade of the package. The proposed code excludes such files from the plib registry results as they already have a legitimate owner.

### CODE REVIEW

File: qfile.c [hunk 1]
```diff
@@ -57,11 +60,77 @@ struct qfile_opt_state {
...
+{
+	struct qfile_opt_state *state = priv;
+
+	int fd_plibreg;
+        FILE *fp_plibreg;
+	struct stat cst;
+
+	/* Open plibreg */
+        fp_plibreg = NULL;
+	fd_plibreg = open("/var/lib/portage/preserved_libs_registry", O_RDONLY|O_CLOEXEC, 0);
```
Comment (grobian) [inline qfile.c]: this needs EPREFIX prepended here one way or the other, I suspect there's actually some defines that point to locations like these, we should use one

File: qfile.c [hunk 2]
```diff
@@ -57,11 +60,77 @@ struct qfile_opt_state {
...
+			continue;
+		if (non_orphans && non_orphans[i])
+			continue;
+		if (results[i] == 1)
+			continue;
+
+		snprintf(file, sizeof(file), "%s/%s", dir_names[i], base_names[i]);
+
+		while (getline(&line, &len, fp_plibreg) != -1)
+			if (strstr(line, file) != NULL) {
```
Comment (grobian) [inline qfile.c]: Is this going to respect the -b flag?

File: qfile.c [hunk 3]
```diff
@@ -57,11 +60,77 @@ struct qfile_opt_state {
...
+
+        qfile_args_t *args = &state->args;
+        char **base_names = args->basenames;
+        char **dir_names = args->dirnames;
+        short *non_orphans = args->non_orphans;
+        int *results = args->results;
+	char file[_Q_PATH_MAX];
+	char *line = NULL;
+	size_t len = 0;
+	int found = 0;
```
Comment (grobian) [inline qfile.c]: can you move all declarations to the top, please?

File: qfile.c (hunk 1 above)
Comment (sbesl) [reply qfile.c]: Fixed

File: qfile.c (hunk 3 above)
Comment (sbesl) [reply qfile.c]: Fixed

File: qfile.c (hunk 2 above)
Comment (sbesl) [reply qfile.c]: Thx, no, it didn't. Fixed by only putting dir_names[i] in there if it isn't empty.

File: qfile.c (hunk 1 above)
Comment (sbesl) [reply qfile.c]: Ah, the change uses eprefix which is defined in main.c. This worked on v0.74 but not here on git master. Something changed in the relationship between the applet and main.c. Will look into it. Happy to receive pointers if you have any.

Comment (grobian) [reply qfile.c]: qmerge.c:               return qmerge_add_set_file(CONFIG_EPREFIX, "/var/lib/portage",

wouldn't that work?

File: qfile.c (hunk 2 above)
Comment (grobian) [reply qfile.c]: honestly I don't know what I was talking about here, there is no such thing as a `-b` flag in today's qfile.

**Comment by grobian** [comment]
I'm affraid you'll have to force push or something to get only your commits to show up here, it's hard to review since all of the hundreds of changes you pulled in to update also show up in che changes.

File: qfile.c (hunk 2 above)
Comment (sbesl) [reply qfile.c]: -b used to be needed if only basename was provided, this became default behaviour after 5531d5c
from the looks of it this change is needed, but I can't test as my system is on -0.74

Comment (sbesl) [reply qfile.c]: On -0.74 the change makes -b work though :D

File: qfile.c (hunk 1 above)
Comment (sbesl) [reply qfile.c]: Yep, that works.

**Comment by sbesl** [comment]
I'll create a new pull request without all those commits

**Pull request closed without merging**